engine/src/ast/function_expr.rs

```diff
@@ -104,6 +103,16 @@ impl<'s> FunctionCallExpr<'s> {
     }
 }
 
+fn incompatible_number_arguments_err<'i>(function: &Function, input: &'i str) -> LexError<'i> {
+    (
+        LexErrorKind::IncompatibleNumberArguments {
```

Comment:
Idk. Technically this name is almost correct in that it's the number of arguments that is incompatible, but then it doesn't state incompatible with what...

Should it be called maybe `InvalidArgumentCount` instead?

Reply:
Renamed.

Reply:
I was more talking about renaming to `IncompatibleNumberParams` but i guess because this happens when parsing a function call, this name remains sensible.